tests/test_file_writer.py

```diff
@@ -486,62 +487,86 @@ def test_FileWriterProcess__start_recording__sets_stop_recording_timestamp_to_no
+def test_FileWriterProcess__stop_recording_sets_stop_recording_timestamp_to_timepoint_in_communication_and_communicates_successful_receipt_and_sets_is_recording_to_false__and_start_recording_clears_stop_timestamp_and_finalization_statuses(
     four_board_file_writer_process,
 ):
-    # should maybe be replaced by a broader test that a recording can be started and stopped twice successfully...
     (
+        # pylint: disable=duplicate-code
         file_writer_process,
         board_queues,
+        # pylint: disable=duplicate-code
         from_main_queue,
         to_main_queue,
         _,
+        # pylint: disable=duplicate-code
         _,
     ) = four_board_file_writer_process
 
     expected_well_idx = 0
     this_command = copy.deepcopy(GENERIC_START_RECORDING_COMMAND)
-    this_command["timepoint_to_begin_recording_at"] = 0
+    this_command["timepoint_to_begin_recording_at"] = 440000
     this_command["active_well_indices"] = [expected_well_idx]
     put_object_into_queue_and_raise_error_if_eventually_still_empty(
-        this_command, from_main_queue, timeout_seconds=QUEUE_CHECK_TIMEOUT_SECONDS
+        this_command, from_main_queue
     )
     invoke_process_run_and_check_errors(file_writer_process)
 
-    a_data_packet = {
-        "well_index": expected_well_idx,
+    data_packet = {
         "is_reference_sensor": False,
-        "data": np.array([[0], [1]], dtype=np.int32),
+        "well_index": expected_well_idx,
+        "data": np.array([[440000], [0]], dtype=np.int32),
     }
     put_object_into_queue_and_raise_error_if_eventually_still_empty(
-        a_data_packet, board_queues[0][0], timeout_seconds=QUEUE_CHECK_TIMEOUT_SECONDS
+        data_packet, board_queues[0][0]
     )
     invoke_process_run_and_check_errors(file_writer_process)
 
     stop_timestamps = file_writer_process.get_stop_recording_timestamps()
 
     assert stop_timestamps[0] is None
 
+    this_command = copy.deepcopy(GENERIC_STOP_RECORDING_COMMAND)
+    this_command["timepoint_to_stop_recording_at"] = 2968000
     put_object_into_queue_and_raise_error_if_eventually_still_empty(
-        GENERIC_STOP_RECORDING_COMMAND,
-        from_main_queue,
-        timeout_seconds=QUEUE_CHECK_TIMEOUT_SECONDS,
+        this_command, from_main_queue
     )
     invoke_process_run_and_check_errors(file_writer_process)
 
-    assert stop_timestamps[0] == 302412 * 125
+    assert stop_timestamps[0] == 2968000
 
-    to_main_queue.get(
-        timeout=QUEUE_CHECK_TIMEOUT_SECONDS
-    )  # pop off the initial receipt of start command message
-    assert_queue_is_eventually_not_empty(to_main_queue)
-    comm_to_main = to_main_queue.get(timeout=QUEUE_CHECK_TIMEOUT_SECONDS)
+    confirm_queue_is_eventually_of_size(
+        to_main_queue, 2, timeout_seconds=QUEUE_CHECK_TIMEOUT_SECONDS
+    )
+    to_main_queue.get_nowait()  # pop off the initial receipt of start command message
```

Comment:
for future, we're stopping using `get_nowait` as it can cause sporadic test failures in CI.  prefer `get(timeout=QUEUE_CHECK_TIMEOUT_SECONDS)`